crossfire/__init__.py

```diff
@@ -68,19 +70,23 @@ def flatten(data, nested_columns=None):
     nested_columns = set(nested_columns or NESTED_COLUMNS)
     if not nested_columns.issubset(NESTED_COLUMNS):
         raise NestedColumnError(nested_columns)
-    if isinstance(data, DataFrame) and not data.empty:
-        keys = set(data.columns) & nested_columns
-        for key in keys:
-            data = concat(
-                (
-                    data.drop(key, axis=1),
-                    data.apply(_flatten_df, args=(key,), axis=1),
-                ),
-                axis=1,
-            )
-
-        return data
-    elif not data:
+    if HAS_PANDAS and isinstance(data, DataFrame):
+        if data.empty:
+            return data
+        else:
+            keys = set(data.columns) & nested_columns
+            for key in keys:
+                data = concat(
+                    (
+                        data.drop(key, axis=1),
+                        data.apply(_flatten_df, args=(key,), axis=1),
+                    ),
+                    axis=1,
+                )
+
+            return data
+
+    if not data:
         return data
```

Comment:
I know, still not being an early return... but I coun't figure out a way to put this in the beggining with repeating some validations...

Reply:
You can use the `HAS_PANDAS` to know whether to check for an empty list or an empty `DataFrame`, can't you?

Reply:
Yes. I believe that i what I am doing [here](https://github.com/FelipeSBarros/crossfire/pull/103#discussion_r1503256591). Right?